security-analysis/security-analysis-default/src/main/java/com/powsybl/security/impl/DefaultSecurityAnalysis.java

```diff
@@ -284,4 +292,22 @@ private ThreeWindingsTransformerResult createThreeWindingsTransformerResult(Thre
                 threeWindingsTransformer.getLeg3().getTerminal().getP(), threeWindingsTransformer.getLeg3().getTerminal().getQ(), threeWindingsTransformer.getLeg3().getTerminal().getI());
     }
 
+    private void checkPreContingencyViolations(Network network, Consumer<LimitViolation> consumer) {
+        if (violationDetector != null) {
+            violationDetector.checkAll(network, consumer);
+        } else {
+            LimitViolationDetection.checkAll(network, EnumSet.allOf(LoadingLimitType.class), 1., consumer);
+        }
+    }
+
+    protected void checkPostContingencyViolations(Contingency contingency, Network network, Consumer<LimitViolation> consumer) {
+        if (violationDetector != null) {
+            violationDetector.checkAll(contingency, network, consumer);
+        } else {
+            // TODO For now, contingencies are ignored in the default security analysis
```

Comment:
Maybe remove as already discussed the `TODO For now` and `This should change to fully support LimitReductions`, and add something like `This blocks a full support of LimitReductions`. Indeed we won't probably change this behaviour in the future.

Reply:
I removed the `TODO` and added a clearer comment.